File: app/src/main/java/com/fox/training/intentAndViewPager/viewpager/fragment/TopMusicFragment.java
package com.fox.training.intentAndViewPager.viewpager.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.fox.training.R;
import com.fox.training.recyclerview.adapters.TopMusicAdapter;
import com.fox.training.recyclerview.data.network.response.Music;

import java.util.ArrayList;

public class TopMusicFragment extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View mRootView = inflater.inflate(R.layout.fragment_topmusic, container, false);
        RecyclerView mTopMusicRecyclerView = mRootView.findViewById(R.id.rvTopMusic);
        ArrayList<Music> mListMusic = new ArrayList<>();

        mListMusic.add(new Music(R.drawable.img_peashooter,
                getString(R.string.peashooter_song_name), getString(R.string.peashooter_song_author)));
        mListMusic.add(new Music(R.drawable.img_sunflower,
                getString(R.string.sunflower_song_name), getString(R.string.sunflower_song_author)));
        mListMusic.add(new Music(R.drawable.img_cherrybomb,
                getString(R.string.cherry_bomb_song_name), getString(R.string.cherry_bomb_song_author)));

        TopMusicAdapter mMusicAdapter = new TopMusicAdapter(getContext(), mListMusic);
        mTopMusicRecyclerView.setLayoutManager(new GridLayoutManager(getContext(), 1));
        mTopMusicRecyclerView.setAdapter(mMusicAdapter);
        return mRootView;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
    }

}

File: app/src/main/java/com/fox/training/ui/main/fragment/TopMusicFragment.kt
package com.fox.training.ui.main.fragment

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.fox.training.data.network.response.Music
import com.fox.training.databinding.FragmentTopmusicBinding
import com.fox.training.ui.main.adapter.TopMusicAdapter


class TopMusicFragment : Fragment() {

    private var musicList = mutableListOf<Music>()
    private lateinit var binding: FragmentTopmusicBinding
    private lateinit var viewModel: MusicViewModel

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentTopmusicBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        viewModel = ViewModelProvider(this).get(MusicViewModel::class.java)
        setupViews()
        setData()
    }

    private fun setupViews() = binding.run {
        recycleViewTopMusic.run {
            layoutManager = LinearLayoutManager(context)
            adapter = TopMusicAdapter(musicList) {

            }
        }
    }

    @SuppressLint("NotifyDataSetChanged")
    private fun setData() {
        viewModel.listMusicLiveData.observe(viewLifecycleOwner) {
            musicList.run {
                clear()
                addAll(it)
            }
            binding.recycleViewTopMusic.adapter?.notifyDataSetChanged()
        }
        viewModel.getTopMusic()
    }
}